src/components/course/EnrollButton.jsx

```diff
@@ -131,176 +71,124 @@ export default function EnrollButton() {
-              isUserEnrolled={isUserEnrolled}
-              pacingType={pacingType}
-              start={start}
-              userEntitlements={userEntitlements}
-            />
-          </a>
-        );
-      } if (enrollmentUrl) {
-        return (
-          <>
-            <Button
-              className={classNames('btn', enrollLinkClass)}
-              onClick={() => setIsModalOpen(true)}
-              variant="success"
-            >
-              <EnrollButtonLabel
-                activeCourseRun={activeCourseRun}
-                availability={availability}
-                courseUuid={courseUuid}
-                isCourseStarted={isCourseStarted}
-                isEnrollable={isEnrollable}
-                isUserEnrolled={isUserEnrolled}
-                pacingType={pacingType}
-                start={start}
-                userEntitlements={userEntitlements}
-              />
-            </Button>
-            <EnrollModal
-              isModalOpen={isModalOpen}
-              setIsModalOpen={setIsModalOpen}
-              offersCount={offersCount}
-              courseHasOffer={!!findOfferForCourse(offers, catalogList)}
-              enrollmentUrl={enrollmentUrl}
-            />
-          </>
-        );
-      }
+  const enrollLinkClass = 'btn-block rounded-0 py-2';
+
+  if (!userEnrollment && isEnrollable) {
+    // enroll with a subscription license
+    if (enrollmentUrl && subscriptionLicense) {
       return (
-        <DefaultEnrollCta
-          className={classNames(enrollLinkClass, 'disabled')}
-          variant="success"
+        <EnrollButtonCta
+          as="a"
+          className={classNames('btn', 'btn-success', enrollLinkClass)}
+          href={enrollmentUrl}
         />
       );
     }
-
-    if (!isUserEnrolled && !isEnrollable) {
+    // enroll with an offer (code)
+    if (enrollmentUrl) {
       return (
-        <div className="alert alert-secondary text-center rounded-0">
-          <EnrollButtonLabel
-            activeCourseRun={activeCourseRun}
-            availability={availability}
-            courseUuid={courseUuid}
-            isCourseStarted={isCourseStarted}
-            isEnrollable={isEnrollable}
-            isUserEnrolled={isUserEnrolled}
-            pacingType={pacingType}
-            start={start}
-            userEntitlements={userEntitlements}
+        <>
+          <EnrollButtonCta
+            className={classNames('btn', enrollLinkClass)}
+            onClick={() => setIsModalOpen(true)}
+            variant="success"
+          />
+          <EnrollModal
+            isModalOpen={isModalOpen}
+            setIsModalOpen={setIsModalOpen}
+            offersCount={offersCount}
+            courseHasOffer={!!findOfferForCourse(offers, catalogList)}
+            enrollmentUrl={enrollmentUrl}
           />
-        </div>
+        </>
       );
     }
+    // cannot enroll without a valid enrollment url, so render a disabled button.
```

Comment:
We should talk to Ange about this at some point; I'd rather not display a button at all if you can't enroll. It's not super clear to a new user that the button is disabled.

Reply:
Yeah, I noted this as a follow-up question as well. For this PR, let's keep it as is. The disabled enroll button is definitely an edge case that really shouldn't ever happen.

Reply:
Agreed, though with the feature flag, it'll happen for codes right now. But that's hopefully going away very soon. :)